app/services/set_school_local_authority_district.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class SetSchoolLocalAuthorityDistrict < BaseService
+  def call
+    if school.local_authority_district&.code != administrative_district_code
+      ActiveRecord::Base.transaction do
+        school.school_local_authority_districts.latest.first&.update!(end_year: start_year)
+
+        SchoolLocalAuthorityDistrict.create!(school: school,
+                                             local_authority_district: la_district,
+                                             start_year: start_year)
+        update_participants_sparsity_flag!
+      end
+    end
+  end
+
+private
+
+  attr_reader :administrative_district_code, :school, :start_year
+
+  def initialize(school:, administrative_district_code: nil, start_year: Time.zone.now.year)
+    @school = school
+    @administrative_district_code = administrative_district_code || school.administrative_district_code
+    @start_year = start_year
+  end
+
+  def update_participants_sparsity_flag!
+    unless school.partnerships.any?
```

Comment:
I would guess this wants an `unchallenged` or maybe an `active`, but I'm not sure which

Reply:
I'm not sure the only steer was if there's a partnership leave sparsity alone, so thought it safest to just test whether there was or had been any?

Reply:
I'm going to remove updating the participants sparsity as it should only be set at the point in time they are added.